dwh/indicator_lib.py

```diff
@@ -82,14 +82,48 @@ def calc_TX_PVLS(patient_agg_obs: pandas.DataFrame, VL_code: str,
   if end_date_str:
     end_date = datetime.strptime(end_date_str, '%Y-%m-%d')
   temp_df = patient_agg_obs[(patient_agg_obs['code'] == VL_code)].copy()
+  # separate value from date
+  temp_df[['vl_max_date', 'latest_vl_value']] = temp_df.max_date_value.str.split('_SeP_', expand=True)
+  temp_df['latest_vl_value'] = temp_df['latest_vl_value'].astype(float)
   # Note the above copy is used to avoid setting a new column on a slice next:
-  temp_df['sup_VL'] = (temp_df['max_value'] < failure_threshold)
+  temp_df['sup_VL'] = (temp_df['latest_vl_value'] < failure_threshold)
   temp_df['buckets'] = temp_df.apply(
       lambda x: _agg_buckets(x.birthDate, x.gender, datetime.today()), axis=1)
```

Comment:
Instead of `today()` we should use `end_date`, I think. This is an error I have introduced in [this PR](https://github.com/GoogleCloudPlatform/openmrs-fhir-analytics/pull/178). Can you please fix it here and in the next function?

Reply:
Noted